### PR TITLE
Fix overfetch of data on user login

### PR DESCRIPTION
**Pull request by Cloud11PL**
I want to merge this because it removes unused queries from the user query such as addresses and metadata on login. 

PR to be tested with this change: https://github.com/saleor/saleor-dashboard/pull/1909

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-5964 Log in to dashboard overfetch data.](https://app.clickup.com/t/2549495/SALEOR-5964)

**Review by orzechdev — COMMENTED** [review]
Please update tests as they fail, probably due to the incorrect polly mocks - some of them should have probably removed fields `metadata`, `addresses`, `defaultBillingAddress`,  `defaultShippingAddress`, as they are no longer requested via queries and mutations.

File: src/core/auth.ts [hunk 1]
```diff
@@ -181,9 +182,10 @@ export const auth = ({
   apolloClient: client,
   channel,
 }: SaleorClientMethodsProps): AuthSDK => {
-  const login: AuthSDK["login"] = opts => {
+  const login: AuthSDK["login"] = (opts, includeDetails = false) => {
```
[the remark this answers is not in the document]
Comment (Cloud11PL) [reply src/core/auth.ts:185]: @patrys To match other conditional queries in sdk - example: https://github.com/saleor/saleor-sdk/blob/main/src/core/auth.ts#L256

[the remark this answers is not in the document]
Comment (patrys) [reply src/core/auth.ts:185]: And why is it not a part of `opts` in other places?

File: src/core/auth.ts [hunk 2]
```diff
@@ -84,9 +85,12 @@ export interface AuthSDK {
    * Authenticates user with email and password.
    *
    * @param opts - Object with user's email and password.
+   * @param includeDetails - Whether to fetch user details. Default is true.
```
Comment (orzechdev) [inline src/core/auth.ts:88]: includeDetails description should not be described as separate param, instead as param in opts param

File: src/core/auth.ts [hunk 3]
```diff
@@ -84,9 +85,12 @@ export interface AuthSDK {
...
    * @returns Promise resolved with CreateToken type data.
    */
-  login: (opts: LoginOpts) => Promise<LoginResult>;
+  login: (
+    opts: LoginOpts & { includeDetails?: boolean }
```
Comment (orzechdev) [inline src/core/auth.ts:92]: Can we add `{ includeDetails?: boolean }` to `LoginOpts`? Like:
```ts
type LoginOpts = MutationTokenCreateArgs & { includeDetails?: boolean };
```

**Review by bmigirl — APPROVED** [review]

**Review by orzechdev — APPROVED** [review]

**Review by jwm0 — APPROVED** [review]

**Pull request merged**